src/Bot/commands/commandsReplies.ts

```diff
@@ -44,3 +45,24 @@ export const lookupReply = async (interaction: CommandInteraction) => {
     await interaction.editReply({ content: 'We could not find the user.' })
   }
 }
+
+export const assistantAskReply = async (interaction: CommandInteraction) => {
+  const questionArg = interaction.options.getString('question')
+
+  if (!questionArg) {
+    await interaction.reply({ content: 'You need to provide a question.', ephemeral: true })
+    return
+  }
+
+  try {
+    // Sometimes, the model takes more than 3 seconds to respond, so we need to defer the reply
+    // to let the user know that the bot is processing the request and it won't be rejected
+    await interaction.deferReply()
```

Comment:
Don't await the deferReply, you can just fire off the request to deferReply and move on to making the gpt call.

```js
interaction.deferReply()
const { completion } = await sendPrompt(questionArg)
```

Reply:
I awaited because that's what they did in the docs: https://discordjs.guide/slash-commands/response-methods.html#deferred-responses

I'm not sure why, but if for any reason it needs to be awaited in order for other functions to not break, then we should `await` it.

Reply:
Then await it after the prompt request.

```js
const promptPromise = sendPrompt(questionArg)
await interaction.deferReply()
const { completion } = await promptPromise
```

Reply:
You're right, great point. We need to get the completion as fast as possible.